docs/_plugins/uid-permalinks.rb

```diff
@@ -0,0 +1,64 @@
+
+
+class PageUIDCollisionException < StandardError
+  def initialize(msg="Page UID hashes collided.")
+    super(msg)
+  end
+end
+
+module Jekyll
```

Comment:
why is this file on the developer site and not part of the doc generator? I think we should keep any custom business logic in the developer site to a minimum, ideally zero. Should be pure content!

Reply:
Moved the plugin to `tk-doc-generator` and added std header and docstrings.